testutil/integration/common/factory/factory.go

```diff
@@ -38,6 +39,16 @@ type TxFactory interface {
 	// FundAccountWithBaseDenom funds the given account with the given amount of the network's
 	// base denomination.
 	FundAccountWithBaseDenom(sender keyring.Key, receiver sdktypes.AccAddress, amount sdkmath.Int) error
+
+	// Delegate is a method to create and broadcast a MsgDelegate
+	Delegate(delegatorPriv cryptotypes.PrivKey, validatorAddr string, amount sdk.Coin) error
+
+	// SetWithdrawAddress is a method to create and broadcast a MsgSetWithdrawAddress
+	SetWithdrawAddress(delegatorPriv cryptotypes.PrivKey, withdrawerAddr sdk.AccAddress) error
+	// WithdrawDelegationRewards is a method to create and broadcast a MsgWithdrawDelegationRewards
+	WithdrawDelegationRewards(delegatorPriv cryptotypes.PrivKey, validatorAddr string) error
+	// WithdrawValidatorCommission is a method to create and broadcast a MsgWithdrawValidatorCommission
+	WithdrawValidatorCommission(validatorPriv cryptotypes.PrivKey) error
```

Comment:
this interface should be kept as lean as possible - I wanted to remove e.g. the `FundAccount` stuff on another PR too

Reply:
why not add these methods that are common to all cosmos chains?

Reply:
refactored based on comments